docs/aws/audit/esmonitoring/rules/es_https_only_remediation.mdx

```diff
@@ -1,6 +1,245 @@
+
+```python
+# Make a request to verify HTTPS configuration
+try:
+    response = es.info()
+    print("Elasticsearch is configured to use HTTPS.")
+    print(response)
+except Exception as e:
+    print("Error: Elasticsearch is not configured to use HTTPS.")
+    print(e)
+```
+
+### Summary
+1. Install the `elasticsearch` library.
+2. Configure the Elasticsearch client to use HTTPS.
+3. Enforce HTTPS in the Elasticsearch configuration using the API.
+4. Verify the HTTPS configuration by making a request to the cluster.
+
+By following these steps, you can ensure that your Elasticsearch cluster uses HTTPS only, thereby enhancing the security of your data in transit.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the Amazon Elasticsearch Service (Amazon ES) dashboard.
+3. In the navigation pane, choose "Domains".
+4. In the "Domains" list, select the Elasticsearch domain that you want to examine.
+5. In the "Domain overview" section, check the "Domain endpoint" value. If the protocol is not HTTPS, then the selected Amazon Elasticsearch domain is not using secure communication (HTTPS) for all the traffic to and from the domain.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can download it from the official AWS website and configure it using the "aws configure" command. You will be prompted to provide your AWS Access Key ID, Secret Access Key, Default region name, and Default output format.
+
+2. Once the AWS CLI is set up, you can use the "describe-elasticsearch-domain-config" command to retrieve the configuration information of the specified Elasticsearch domain. The command syntax is as follows:
+
+   ```
+   aws es describe-elasticsearch-domain-config --domain-name <your_domain_name>
+   ```
+
+   Replace `<your_domain_name>` with the name of your Elasticsearch domain.
+
+3. The output of the command will be in JSON format. You need to check the "DomainConfig" section in the output. Under this section, look for the "ElasticsearchClusterConfig" field. This field contains the configuration settings for the Elasticsearch domain.
+
+4. Specifically, you need to check the "TLSSecurityPolicy" field under the "DomainEndpointOptions" section. If the value of this field is not "Policy-Min-TLS-1-2-2019-07", it means that the Elasticsearch domain is not using HTTPS only.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Import the necessary libraries: You will need the boto3 library for AWS services and the requests library to make HTTP requests.
+
+```python
+import boto3
+import requests
+```
+
+2. Create a session using your AWS credentials and initialize the ElasticSearch client:
+
+```python
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>